crates/provider-wadm/wit/wadm/manifest.wit

```diff
@@ -0,0 +1,91 @@
+interface oam-types {
+    // The overall structure of an OAM manifest.
+    record oam-manifest {
+        api-version: string,
+        kind: string,
+        metadata: metadata,
+        spec: specification,
+    }
+
+    // Metadata describing the manifest
+    record metadata {
+        name: string,
+        annotations: list<tuple<string, string>>,
+        labels: list<tuple<string, string>>,
+    }
+
+    // The specification for this manifest
+    record specification {
+        components: list<component>,
+    }
+
+    // A component definition
+    record component {
+        name: string,
+        properties: properties,
+        traits: option<list<trait>>,
```

Comment:
I think this might be optional in the OAM spec, but we definitely use it. Might just be easier to make this a `list<trait>` without the option